src/Raven.Server/Documents/ETL/Providers/RelationalDatabase/Common/RelationalWriters/RelationalDatabaseWriterBase.cs

```diff
@@ -0,0 +1,571 @@
+    }
+
+    public List<Func<DbParameter, string, bool>> GenerateStringParsers()
+    {
+        return new List<Func<DbParameter, string, bool>>
+        {
+            (colParam, value) =>
+            {
+                if (char.IsDigit(value[0]))
+                {
+                    if (DateTime.TryParseExact(value, DefaultFormat.OnlyDateTimeFormat, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind,
+                            out DateTime dateTime))
+                    {
+                        colParam.Value = dateTime;
+                        return true;
+                    }
+                }
+
+                return false;
+            },
+            (colParam, value) =>
+            {
+                if (char.IsDigit(value[0]))
+                {
+                    if (DateTimeOffset.TryParseExact(value, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture,
+                            DateTimeStyles.RoundtripKind, out DateTimeOffset dateTimeOffset))
+                    {
+                        colParam.Value = dateTimeOffset;
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        };
+    }
+
+    public DbCommand GetInsertCommand(string tableName, string pkName, ToRelationalDatabaseItem itemToReplicate)
+    {
+        var cmd = CreateCommand();
+        
+        var sb = new StringBuilder("INSERT INTO ")
+            .Append(GetTableNameString(tableName))
+            .Append(" (")
+            .Append(_commandBuilder.QuoteIdentifier(pkName))
+            .Append(", ");
+        foreach (var column in itemToReplicate.Columns)
+        {
+            if (column.Id == pkName)
+                continue;
+            sb.Append(_commandBuilder.QuoteIdentifier(column.Id)).Append(", ");
+        }
+
+        sb.Length = sb.Length - 2;
+
+        var pkParam = cmd.CreateParameter();
+        pkParam.ParameterName = GetParameterNameForDbParameter(pkName);
+
+        SetPrimaryKeyParamValue(itemToReplicate, pkParam);
+        cmd.Parameters.Add(pkParam);
+
+        var afterIntoSyntax = GetPostInsertIntoStartSyntax(itemToReplicate);
+
+        sb.Append($") {afterIntoSyntax}");
+
+        sb.Append(GetParameterNameForCommandString(pkName, false)).Append(", ");
+
+        foreach (var column in itemToReplicate.Columns)
+        {
+            if (column.Id == pkName)
+                continue;
+
+            var colParam = cmd.CreateParameter();
+            colParam.ParameterName = GetParameterNameForDbParameter(column.Id);
+            SetParamValue(colParam, column, _stringParserList);
+            EnsureParamTypeSupportedByDbProvider(colParam);
+
+            cmd.Parameters.Add(colParam);
+            sb.Append(GetParameterNameForCommandString(column.Id, column.IsArrayOrObject)).Append(", ");
+        }
+
+        sb.Length = sb.Length - 2;
+        var endSyntax = GetPostInsertIntoEndSyntax(itemToReplicate);
```

Comment:
Instead of having `GetPostInsertIntoStartSyntax / GetPostInsertIntoEndSyntax` maybe we should introduce the notion of scope.

Reply:
What about using ` SqlCommandBuilder` / `SnowflakeDbCommandBuilder `?

Reply:
imo such builders are unnecessary here, it won't differ too much from the method, which is a builder itself, and is easy to debug 

However, I could merge these methods into `GetPostInsertIntoSyntaxScope` or smth similar